plugins/auth-backend/src/providers/gitlab/provider.ts

```diff
@@ -145,42 +170,141 @@ export class GitlabAuthProvider implements OAuthHandlers {
       this._strategy,
       accessToken,
     );
-    const { id, profile } = transformProfile(fullProfile);
 
-    return {
-      profile,
+    return this.handleResult({
+      fullProfile,
+      params,
+      accessToken,
+      refreshToken: newRefreshToken,
+    });
+  }
+
+  private async handleResult(result: OAuthResult): Promise<OAuthResponse> {
+    const transformedResult = this.transformResult(result);
```

Comment:
The things going on in `transformResult` should go into the `authHandler`, as that's the piece responsible for transforming the full passport profile into our representation. To slim down the implementation of the default handler I think we could move the `avatarUrl` workaround that was previously in `transformProfile` straight into the common `makeProfileInfo` implementation.